lib/simulator-xcode-11.4.js

```diff
@@ -107,67 +89,12 @@ class SimulatorXcode11_4 extends SimulatorXcode11 {
 
   /**
    * @override
-   * Sets the permissions for the particular application bundle.
-   *
-   * @param {string} bundleId - Application bundle identifier.
-   * @param {Object} permissionsMapping - A mapping where kays
-   * are service names and values are their corresponding status values.
-   * The following keys are supported:
-   * - all: Apply the action to all services.
-   * - calendar: Allow access to calendar.
-   * - contacts-limited: Allow access to basic contact info.
-   * - contacts: Allow access to full contact details.
-   * - location: Allow access to location services when app is in use.
-   * - location-always: Allow access to location services at all times.
-   * - photos-add: Allow adding photos to the photo library.
-   * - photos: Allow full access to the photo library.
-   * - media-library: Allow access to the media library.
-   * - microphone: Allow access to audio input.
-   * - motion: Allow access to motion and fitness data.
-   * - reminders: Allow access to reminders.
-   * - siri: Allow use of the app with Siri.
-   * The following values are supported:
-   * - yes: To grant the permission
-   * - no: To revoke the permission
-   * - unset: To reset the permission
-   * @throws {Error} If there was an error while changing permissions.
    */
   async setPermissions (bundleId, permissionsMapping) {
-    log.debug(`Setting access for '${bundleId}': ` +
-      JSON.stringify(permissionsMapping, null, 2));
-    const nonNativePerms = {};
-    for (let [permName, access] of _.toPairs(permissionsMapping)) {
-      if (permName === 'medialibrary') {
-        permName = 'media-library';
-      } else if (permName === 'location' && _.toLower(access) === 'always') {
-        permName = 'location-always';
-      } else if (!NATIVE_SIMCTL_PERMISSIONS.includes(permName)) {
-        nonNativePerms[permName] = access;
-        continue;
-      }
+    return await super.setPermissions(bundleId, permissionsMapping);
 
-      switch (_.toLower(access)) {
-        case 'yes':
-        case 'inuse':
-        case 'always':
-          await this.simctl.grantPermission(bundleId, permName);
-          break;
-        case 'no':
-        case 'never':
-          await this.simctl.revokePermission(bundleId, permName);
-          break;
-        case 'unset':
-          await this.simctl.resetPermission(bundleId, permName);
-          break;
-        default:
-          throw new Error(`Unknown access value: ${access}`);
-      }
-    }
-    if (!_.isEmpty(nonNativePerms)) {
-      log.info(`The following permissions have not been recognized as native: ` +
-        JSON.stringify(nonNativePerms, null, 2));
-      await super.setPermissions(bundleId, nonNativePerms);
-    }
+    // TODO: Switch to `simctl privacy` call after Apple
+    // fixes the command (https://github.com/appium/appium/issues/14355)
```

Comment:
It would be good to leave this revert PR, too

Reply:
added